### ISSUE TITLE
Failure to parse cast + fold expression

### ISSUE THREAD

**Issue by rupprecht** [llvm/llvm-project#56864]
Clang fails to parse a parameter pack expansion after a C-style cast:

```cpp
template <class... T>
void Test(std::tuple<T...>) {
    (void) (F<T>(), ...);
}
```

Clang fails with:

```console
<source>:8:21: error: expected expression
    (void) (F<T>(), ...);
                    ^
```

I believe, but I'm not sure, this may be a bug in Clang. Removing `(void)` or adding an extra set of parentheses allows Clang to parse it.

Compiler Explorer link: https://godbolt.org/z/zje9EG4WT

(reported internally)

**Comment by shafik** [comment]
This indeed looks like a clang bug, according to [\[temp.variadic\]p10.2](https://eel.is/c++draft/temp.variadic#10.2) this should expand to

```cpp
(void) F<int>(),(F<int>());
```

which is valid. Note, the `(void)` cast will only apply to the first `F<int>()` and not the rest of the calls.

**Comment by zygoloid** [comment]
Fixed in 73b62f813550b602f189afc3a60b0b39ae89f16d.

**Comment by zygoloid** [comment]
> Note, the `(void)` cast will only apply to the first `F<int>()` and not the rest of the calls.

That's a bug in the standard wording! Filed as https://github.com/cplusplus/CWG/issues/109.
